### PR TITLE
Fix #632 Change the default settings to "useRtmConnect: true"

### LINKED ISSUE #632: Change the default settings to "useRtmConnect: true"
### Description

>The original way to connect to one of our oldest APIs is finally retiring. For existing apps, rtm.start will start behaving exactly like rtm.connect on September 20, 2022. Beginning November 30, 2021, newly created apps and integrations will only be able to use rtm.connect.

https://api.slack.com/changelog/2021-10-rtm-start-to-stop

Until this change is released, please set the following env variable when starting your Hubot app:

```
HUBOT_SLACK_RTM_CLIENT_OPTS='{"useRtmConnect": "true"}'
```

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [x] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/hubot-slack/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request fixes #632 -- refer to the issue for the context

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/hubot-slack/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: slack.coffee [hunk 1]
```diff
@@ -9,9 +9,18 @@ exports.use = (robot) ->
...
+    options.rtm = JSON.parse(process.env.HUBOT_SLACK_RTM_CLIENT_OPTS or '{}')
+    unless options.rtm.useRtmConnect?
+      # The original way to connect to one of our oldest APIs is finally retiring.
+      # For existing apps, rtm.start will start behaving exactly like rtm.connect
+      # on September 20, 2022. Beginning November 30, 2021, newly created apps and integrations
+      # will only be able to use rtm.connect.
+      # https://api.slack.com/changelog/2021-10-rtm-start-to-stop
+      options.rtm.useRtmConnect = true
+  catch e
+    console.error(e)
```
Comment (seratch) [inline slack.coffee:21]: Just for better developer experience. The code has been slightly ignoring the exception raised here.

File: slack.coffee [hunk 2]
```diff
@@ -9,9 +9,18 @@ exports.use = (robot) ->
     # reacts to only messages by insalled workspace users in a shared channel
     installedTeamOnly: process.env.INSTALLED_TEAM_ONLY?
   try
-    options.rtm = JSON.parse(process.env.HUBOT_SLACK_RTM_CLIENT_OPTS)
-  catch
+    options.rtm = JSON.parse(process.env.HUBOT_SLACK_RTM_CLIENT_OPTS or '{}')
```
Comment (seratch) [inline slack.coffee:12]: If you set the env variable to an empty string, the parsing fails. This change improves the behavior in the scenario.

**Comment by seratch** [comment]
I've verified the behavior with an example Hubot app.

**Review by stevengill — APPROVED** [review]
LGTM

**Pull request merged**